modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/TransportRankEvalAction.java

```diff
@@ -73,6 +91,21 @@ protected void doExecute(RankEvalRequest request, ActionListener<RankEvalRespons
             final RankEvalActionListener searchListener = new RankEvalActionListener(listener, qualityTask.getMetric(), querySpecification,
                     partialResults, errors, responseCounter);
             SearchSourceBuilder specRequest = querySpecification.getTestRequest();
+            if (specRequest == null) {
+                Map<String, Object> params = querySpecification.getParams();
+                Script scriptWithParams = new Script(qualityTask.getTemplate().getType(), qualityTask.getTemplate().getLang(),
+                        qualityTask.getTemplate().getIdOrCode(), params);
+                String resolvedRequest = ((BytesReference) (scriptService.executable(scriptWithParams, ScriptContext.Standard.SEARCH)
+                        .run())).utf8ToString();
+                logger.error("RANTRANTRANT" + resolvedRequest);
+                try (XContentParser subParser = XContentFactory.xContent(resolvedRequest).createParser(resolvedRequest)) {
+                    QueryParseContext parseContext = new QueryParseContext(searchRequestParsers.queryParsers, subParser, parseFieldMatcher);
+                    specRequest = SearchSourceBuilder.fromXContent(parseContext, searchRequestParsers.aggParsers,
+                            searchRequestParsers.suggesters, searchRequestParsers.searchExtParsers);
+                } catch (IOException e) {
+                    listener.onFailure(e);
```

Comment:
Just a question: this means we are failing the whole rank_eval request here, no? I think this is okay, as long as this is what you intended.